### PR TITLE
Fix authorization of resources

### PR DESCRIPTION
**Pull request by NielsKSchjoedt**
https://github.com/Apipie/apipie-rails/issues/654

### CODE REVIEW

File: lib/apipie/resource_description.rb [hunk 1]
```diff
@@ -106,6 +106,7 @@ def to_json(method_name = nil, lang = nil)
 
       {
         :doc_url => doc_url,
+        :resource_name => _id,
```
Comment (iNecas) [inline lib/apipie/resource_description.rb]: I wonder if we shouldn't call it `:id`, as we already have the `:name` field, and it might be misleading

**Comment by iNecas** [comment]
Looks good, one comment to resolve

File: lib/apipie/resource_description.rb (hunk 1 above)
Comment (NielsKSchjoedt) [inline lib/apipie/resource_description.rb]:
```suggestion
        :id => _id,
```

File: app/controllers/apipie/apipies_controller.rb [hunk 2]
```diff
@@ -115,20 +115,30 @@ def authorized_doc
...
+        @doc[:docs][:resources].select do |_resource_name, resource|
+          authorize_resource(resource)
         end
       end
 
       new_doc
     end
 
+    def authorize_resource resource
+      if instance_exec(resource[:resource_name], nil, resource, &Apipie.configuration.authorize)
```
Comment (NielsKSchjoedt) [inline app/controllers/apipie/apipies_controller.rb]:
```suggestion
      if instance_exec(resource[:id], nil, resource, &Apipie.configuration.authorize)
```

File: app/controllers/apipie/apipies_controller.rb [hunk 3]
```diff
@@ -115,20 +115,30 @@ def authorized_doc
...
+        resource[:methods] = resource[:methods].select do |m|
+          instance_exec(resource[:resource_name], m[:name], m, &Apipie.configuration.authorize)
```
Comment (NielsKSchjoedt) [inline app/controllers/apipie/apipies_controller.rb]:
```suggestion
          instance_exec(resource[:id], m[:name], m, &Apipie.configuration.authorize)
```

File: lib/apipie/resource_description.rb (hunk 1 above)
Comment (NielsKSchjoedt) [reply lib/apipie/resource_description.rb]: Done

**Comment by iNecas** [comment]
apipie-rails-0.5.15 including this change has just been released

**Pull request merged**